Implement 'dirtiness' of HTMLOptionElement

Ahmad-S792 opened the pull request:
#### a5be9b06de7e5b646102d830bea8c2b149bc9f73
<pre>
Implement &apos;dirtiness&apos; of HTMLOptionElement
<a href="https://bugs.webkit.org/show_bug.cgi?id=258073">https://bugs.webkit.org/show_bug.cgi?id=258073</a>
<a href="https://rdar.apple.com/problem/111096541">rdar://problem/111096541</a>

Reviewed by NOBODY (OOPS!).

Merge: <a href="https://chromium.googlesource.com/chromium/src.git/+/a17cc1eda58ad8270dc56ebafab215e67e6c1bfe">https://chromium.googlesource.com/chromium/src.git/+/a17cc1eda58ad8270dc56ebafab215e67e6c1bfe</a> ,
<a href="https://chromium.googlesource.com/chromium/src.git/+/f67251360e4b76d188bb1523d7a0952c2f449e68">https://chromium.googlesource.com/chromium/src.git/+/f67251360e4b76d188bb1523d7a0952c2f449e68</a> and
<a href="https://chromium.googlesource.com/chromium/src/+/6d095ec7a71eb72e2ad9992d1ca3080f5726f6bf">https://chromium.googlesource.com/chromium/src/+/6d095ec7a71eb72e2ad9992d1ca3080f5726f6bf</a>

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

This patch implements the concept of &apos;dirtiness&apos; as per [1]:

[1] <a href="https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-dirtiness">https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-dirtiness</a>

The flag is referred when &apos;selected&apos; attribute is added or removed.
This flag is set to true when a user updates selectedness explicitly, or
HTMLSelectElement::value setter, selectedIndex setter, or
HTMLOptionElement::selected setter.
The flag is reset to false on resetting `select`.

We need to introduce HTMLOptionElement::setSelectedForBinding because
setSelected is used internally, and the dirtiness should not be changed for the
internal callsites.

Additionally, HTMLOptionElement.prototype.selected setter should
not make an option dirty if it doesn&apos;t change its selectedness
and the option is owned by a select element.

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::attributeChanged):
(WebCore::HTMLOptionElement::setDirty):
(WebCore::HTMLOptionElement::setSelectedForBinding):
* Source/WebCore/html/HTMLOptionElement.h:
* Source/WebCore/html/HTMLOptionElement.idl:
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::optionSelectedByUser):
(WebCore::HTMLSelectElement::updateListBoxSelection):
(WebCore::HTMLSelectElement::setSelectedIndex):
(WebCore::HTMLSelectElement::selectOption):
(WebCore::HTMLSelectElement::restoreFormControlState):
(WebCore::HTMLSelectElement::parseMultipleAttribute):
(WebCore::HTMLSelectElement::reset):
(WebCore::HTMLSelectElement::menuListDefaultEventHandler):
(WebCore::HTMLSelectElement::updateSelectedState):
(WebCore::HTMLSelectElement::typeAheadFind):
* Source/WebCore/html/HTMLSelectElement.h:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-option-element/option-selected-expected.txt: Progression
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a5be9b06de7e5b646102d830bea8c2b149bc9f73

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122097 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32469 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128664 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74191 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5bb6a9aa-e496-4d5e-9d43-98619baf87bb) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123973 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42514 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50393 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92802 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61686 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d167d6c2-064f-48d0-8a65-aae4a81577d0) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125049 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33907 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109338 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73458 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bc85eb7c-9c51-42a2-ade5-efdd406d406c) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32921 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27504 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72156 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103414 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27695 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131421 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49036 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37298 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101367 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49410 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105552 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101237 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46605 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24722 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45778 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48893 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54627 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48363 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51713 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50043 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5d89897e4f573e5735df5a6247e331671d605539. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30802

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4a9ea7f2a3dfef654451d8341abb54d218dec9ff. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30802

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8a81acb174cb2a76660af950946c58fac3d84b9d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30802

karlcow reviewed (commented):
nit: typo in the commit message for "inrorudce"
> We need to inrorudce HTMLOptionElement::setSelectedForBinding because

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/ebf44703f0211ac8be217c17cf0d1e09c2ce1fd1)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58342 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37670 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10826 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61967 "Build is in progress. Recent messages:") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8787 "Build is in progress. Recent messages:Running configuration") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60471 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45306 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8983 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/61967 "Build is in progress. Recent messages:") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/59/builds/8787 "Build is in progress. Recent messages:Running configuration") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60373 "Build is in progress. Recent messages:") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/45306 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/55/builds/10826 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/61967 "Build is in progress. Recent messages:") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/45306 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/55/builds/10826 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7791 "Build is in progress. Recent messages:Running configuration") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/45306 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/55/builds/10826 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63672 "Build is in progress. Recent messages:Running configuration") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2256 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/61/builds/8983 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/63672 "Build is in progress. Recent messages:Running configuration") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2264 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/55/builds/10826 "Build is in progress. Recent messages:Running configuration") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/63672 "Build is in progress. Recent messages:Running configuration") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | </details>

darinadler commented on Source/WebCore/html/HTMLOptionElement.h:
```diff
@@ -51,6 +51,8 @@ class HTMLOptionElement final : public HTMLElement {
 
     WEBCORE_EXPORT bool selected(AllowStyleInvalidation = AllowStyleInvalidation::Yes) const;
     WEBCORE_EXPORT void setSelected(bool);
+    WEBCORE_EXPORT bool selectedForBinding() const;
+    WEBCORE_EXPORT void setSelectedForBinding(bool);
```
If I was doing this, I would probably have renamed the functions for internal use instead of renaming the ones for the bindings; requires a little bit of research about how many call sites those functions have. I’m not sure that "for binding" is the best way to distinguish these. Also, it’s only `setSelected` that needs a second version, and renaming the old `setSelected` would leave us with 3 functions rather than 4, which I think is better. Maybe `setSelectedWithoutSettingIsDirty` might be a good name, or we could find a shorter one. Really depends on how many call sites there are and why exactly it’s important to not set `m_isDirty` at those call sites.

I understand that the Chromium contributor chose this way of doing it.

darinadler commented on Source/WebCore/html/HTMLOptionElement.cpp:
```diff
@@ -238,6 +236,24 @@ void HTMLOptionElement::setSelected(bool selected)
         select->optionSelectionStateChanged(*this, selected);
 }
 
+bool HTMLOptionElement::selectedForBinding() const
+{
+    return selected();
+}
```
This should be inlined in the header.

darinadler commented on Source/WebCore/html/HTMLOptionElement.h:
```diff
@@ -64,7 +66,10 @@ class HTMLOptionElement final : public HTMLElement {
 
     String textIndentedToRespectGroupLabel() const;
 
+    // Update 'selectedness'.
```
This should be done by selecting an appropriate function name, not with a comment. Like name it `setSelectedness` and then the comment isn’t needed.

darinadler commented on Source/WebCore/html/HTMLOptionElement.h:
```diff
@@ -64,7 +66,10 @@ class HTMLOptionElement final : public HTMLElement {
 
+    // Update 'selectedness'.
     void setSelectedState(bool, AllowStyleInvalidation = AllowStyleInvalidation::Yes);
+    // Update 'dirtiness'.
```
This should be done by selecting an appropriate function name, not with a comment. Like name it `setDirtiness` and then the comment isn’t needed.

darinadler commented on Source/WebCore/html/HTMLSelectElement.cpp:
```diff
@@ -1035,14 +1039,18 @@ void HTMLSelectElement::restoreFormControlState(const FormControlState& state)
         size_t foundIndex = searchOptionsForValue(state[0], 0, itemsSize);
-        if (foundIndex != notFound)
+        if (foundIndex != notFound) {
             Ref { downcast<HTMLOptionElement>(*items[foundIndex]) }->setSelectedState(true);
+            Ref { downcast<HTMLOptionElement>(*items[foundIndex]) }->setDirty(true);
```
I suggest a local variable here to make the code less repetitive.

darinadler commented on Source/WebCore/html/HTMLSelectElement.cpp:
```diff
@@ -1052,6 +1060,7 @@ void HTMLSelectElement::restoreFormControlState(const FormControlState& state)
             if (foundIndex == notFound)
                 continue;
             Ref { downcast<HTMLOptionElement>(*items[foundIndex]) }->setSelectedState(true);
+            Ref { downcast<HTMLOptionElement>(*items[foundIndex]) }->setDirty(true);
```
I suggest a local variable here to make the code less repetitive.

Ahmad-S792 committed:
Implement 'dirtiness' of HTMLOptionElement
https://bugs.webkit.org/show_bug.cgi?id=258073
rdar://problem/111096541

Reviewed by NOBODY (OOPS!).

Merge: https://chromium.googlesource.com/chromium/src.git/+/a17cc1eda58ad8270dc56ebafab215e67e6c1bfe ,
https://chromium.googlesource.com/chromium/src.git/+/f67251360e4b76d188bb1523d7a0952c2f449e68 and
https://chromium.googlesource.com/chromium/src/+/6d095ec7a71eb72e2ad9992d1ca3080f5726f6bf

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

This patch implements the concept of 'dirtiness' as per [1]:

[1] https://html.spec.whatwg.org/multipage/form-elements.html#concept-option-dirtiness

The flag is referred when 'selected' attribute is added or removed.
This flag is set to true when a user updates selectedness explicitly, or
HTMLSelectElement::value setter, selectedIndex setter, or
HTMLOptionElement::selected setter.
The flag is reset to false on resetting `select`.

We need to introduce HTMLOptionElement::setSelectedForBinding because
setSelected is used internally, and the dirtiness should not be changed for the
internal callsites.

Additionally, HTMLOptionElement.prototype.selected setter should
not make an option dirty if it doesn't change its selectedness
and the option is owned by a select element.

* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::attributeChanged):
(WebCore::HTMLOptionElement::setDirty):
(WebCore::HTMLOptionElement::setSelectedForBinding):
* Source/WebCore/html/HTMLOptionElement.h:
* Source/WebCore/html/HTMLOptionElement.idl:
* Source/WebCore/html/HTMLSelectElement.cpp:
(WebCore::HTMLSelectElement::optionSelectedByUser):
(WebCore::HTMLSelectElement::updateListBoxSelection):
(WebCore::HTMLSelectElement::setSelectedIndex):
(WebCore::HTMLSelectElement::selectOption):
(WebCore::HTMLSelectElement::restoreFormControlState):
(WebCore::HTMLSelectElement::parseMultipleAttribute):
(WebCore::HTMLSelectElement::reset):
(WebCore::HTMLSelectElement::menuListDefaultEventHandler):
(WebCore::HTMLSelectElement::updateSelectedState):
(WebCore::HTMLSelectElement::typeAheadFind):
* Source/WebCore/html/HTMLSelectElement.h:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-option-element/option-selected-expected.txt: Progression

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a5be9b06de7e5b646102d830bea8c2b149bc9f73. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30802

Ahmad-S792 commented:
Just rebased atm.

darinadler commented:
I did approve these changes 14 months ago, but maybe you wanted to make the changes suggested in my review comments?

Ahmad-S792 commented:
Done here - https://github.com/WebKit/WebKit/commit/300656133429cb9c46126d3ce5829eac61957b10
